antithesis_sdk.h

```diff
@@ -31,7 +33,26 @@ namespace antithesis {
 
     struct JSON;
 
-    typedef std::variant<std::string, bool, char, int, uint64_t, float, double, const char*, JSON> ValueType;
+    typedef std::variant<
+      std::string,
+      bool,
+      char,
+      int,
+      uint64_t,
+      float,
+      double,
+      const char*,
+      JSON,
+      std::vector<std::string>,
+      std::vector<bool>,
+      std::vector<char>,
+      std::vector<int>,
+      std::vector<uint64_t>,
+      std::vector<float>,
+      std::vector<double>,
+      std::vector<const char *>,
+      std::vector<JSON>
+    > ValueType;
```

Comment:
There seems to be a lot of duplication here and below.

What about doing something like this instead:

```C++
    typedef std::variant<std::string, bool, char, int, uint64_t, float, double, const char*, JSON> AtomicValueType;
    typedef std::variant<AtomicValueType, std::vector<AtomicValueType>> ValueType;
```

Then you'd take the existing ``operator<<(ValueType)`` and change it to ``operator<<(AtomicValueType)`` and you'd just have to write the (new) ValueType version, which would have two cases only.

Reply:
Added BasicValueType and JSON_ARRAY